OpenClaw/Engine/Actor/Components/ControllerComponents/HealthComponent.cpp

```diff
@@ -76,7 +76,7 @@ void HealthComponent::AddHealth(int32 health, DamageType damageType, Point impac
         m_CurrentHealth = m_MaxHealth;
     }
 
-    if (oldHealth != m_CurrentHealth)
+    if (oldHealth != m_CurrentHealth || damageType == DamageType_SirenProjectile) // For Siren Projectile - has 0 damage

```

Comment:
This is sooo hacky :D

Reply:
I know :/, but i'm open for suggestions how to do this better

Reply:
Yeah currently there is no built-in way to handle this properly. I'd maybe just add some TODO: comment to tag is as something that's hacky and needs to be fixed somehow.